### ISSUE TITLE
Skillchains occur even when weaponskills miss

### ISSUE THREAD

**Issue by Aslenta** [DerpyProjectGroup/topaz#47]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/DerpyProjectGroup/topaz/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/DerpyProjectGroup/topaz/blob/info/CONTRIBUTING.md)
- [x] I have specified what branch this happens on branch: freedom

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 
If a WS misses, another can close with a WS to form a SC. 
If a WS hits (or misses), another can close with a missing WS and it does 0 damage. 

In retail if either opening or closing WS misses, no SC is triggered.

**Comment by RAIST5150** [comment]
This issue came up in the Canaria discord recently.  Mostly centered around observations of WS chains firing off with 0 damage, but we did't have confirmation on whether those chains spawned from WS landing for damage, landing for 0, or were flat out missed.  While gathering some CP tonight, I took it easy and watched the log for once and managed to capture some examples, which I will past grabs in here marked up to highlight the hit/miss and resulting chains.

This was fairly easy to replicate in an SoA zone at level 99, as it is easy to isolate mobs in the +3-5 level range gap where trust hit rate can be pretty low without any debuffs, and you really only need to offset the ACC/EVA comparison by around 10 or 12 points to ramp up the hit rate to ensure a closing WS lands (when testing this out, 2 or 3 quicksteps flipped it around nicely on 102-105 level mobs).

![Trust_0_WS_Darkness2](https://user-images.githubusercontent.com/73744368/131235422-a8364d2a-8929-44a1-ab25-05ec37cdddd3.jpg)
![Trust_WS_Darkness_DMG2](https://user-images.githubusercontent.com/73744368/131235423-d8f40228-5497-43ef-8943-195af70aa73b.jpg)
